File: urbanairship-react-native/android/src/main/java/com/urbanairship/reactnative/events/RegistrationEvent.java
/* Copyright Urban Airship and Contributors */

package com.urbanairship.reactnative.events;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.facebook.react.bridge.Arguments;
import com.facebook.react.bridge.ReadableMap;
import com.facebook.react.bridge.WritableMap;
import com.urbanairship.reactnative.Event;

/**
 * Registration event.
 */
public class RegistrationEvent implements Event {

    private static final String CHANNEL_REGISTRATION_EVENT = "com.urbanairship.registration";
    private static final String CHANNEL_ID = "channelId";
    private static final String REGISTRATION_TOKEN = "registrationToken";

    private final String channelId;
    private final String registrationToken;

    /**
     * Default constructor.
     *
     * @param channelId The channel ID.
     * @param registrationToken The registration token.
     */
    public RegistrationEvent(@NonNull String channelId, @Nullable  String registrationToken) {
        this.channelId = channelId;
        this.registrationToken = registrationToken;
    }

    @NonNull
    @Override
    public String getName() {
        return CHANNEL_REGISTRATION_EVENT;
    }

    @NonNull
    @Override
    public ReadableMap getBody() {
        WritableMap map = Arguments.createMap();
        map.putString(CHANNEL_ID, channelId);

        if (registrationToken != null) {
            map.putString(REGISTRATION_TOKEN, registrationToken);
        }

        return map;
    }


    @Override
    public boolean isForeground() {
        return true;
    }
}

File: urbanairship-react-native/android/src/main/java/com/urbanairship/reactnative/events/RegistrationEvent.kt
/* Copyright Urban Airship and Contributors */

package com.urbanairship.reactnative.events

import com.facebook.react.bridge.Arguments
import com.facebook.react.bridge.ReadableMap
import com.urbanairship.reactnative.Event

/**
 * Registration event.
 *
 * @param channelId The channel ID.
 * @param registrationToken The registration token.
 */
class RegistrationEvent(private val channelId: String, private val registrationToken: String?) : Event {

    override val name: String
        get() = CHANNEL_REGISTRATION_EVENT

    override val body: ReadableMap
        get() {
            val map = Arguments.createMap()
            map.putString(CHANNEL_ID, channelId)
            registrationToken?.let {
                map.putString(REGISTRATION_TOKEN, it)
            }
            return map
        }

    override val isForeground: Boolean
        get() = true

    companion object {
        private const val CHANNEL_REGISTRATION_EVENT = "com.urbanairship.registration"
        private const val CHANNEL_ID = "channelId"
        private const val REGISTRATION_TOKEN = "registrationToken"
    }
}